### PR TITLE
TypeBuilders refactored.

### PR DESCRIPTION
**Pull request by vbartusevicius**
DateTime now has it's separate DateTimePropertyDefinition

### CODE REVIEW

File: src/Paysera/Bundle/CodeGeneratorBundle/Entity/Definition/FilterTypeDefinition.php [hunk 1]
```diff
@@ -7,17 +7,43 @@ class FilterTypeDefinition extends TypeDefinition
...
+    {
+        return $this->baseFilter;
+    }
+
+    /**
+     * @param bool $baseFilter
+     *
+     * @return $this
+     */
+    public function setBaseFilter($baseFilter)
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Entity/Definition/FilterTypeDefinition.php]: Either param type should be strict, or param type of setExtendedBaseFilter shouldn't. Also `declare(strict_types=1);` is missing

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?php
+
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder.php]: missing `declare(strict_types=1);`

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/MetadataTypeBuilder.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/MetadataTypeBuilder.php]: missing `declare(strict_types=1);`

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/ResultTypeBuilder.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?php
+
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/ResultTypeBuilder.php]: missing `declare(strict_types=1);`

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/SimpleTypeBuilder.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?php
+
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/SimpleTypeBuilder.php]: missing `declare(strict_types=1);`

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/TypeDefinitionBuilderInterface.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+interface TypeDefinitionBuilderInterface
+{
+    /**
+     * @param string $name
+     * @param array $definition
+     *
+     * @return bool
+     */
+    public function supports(string $name, array $definition);
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/TypeDefinitionBuilder/TypeDefinitionBuilderInterface.php]: annotations aren't necessary if return type is declared in method itself `public function supports(string $name, array $definition) : bool;`

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/DefinitionDecorator.php [hunk 7]
```diff
@@ -15,13 +15,15 @@
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/DefinitionDecorator.php]: Only `use Raml\ApiDefinition;` and `use Paysera\Bundle\CodeGeneratorBundle\Entity\Definition\ApiDefinition as DecoratedDefinition;` are required

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/PropertyDefinitionBuilder.php [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        'date-only',
+        'time-only',
+    ];
+
+    /**
+     * @param string $name
+     * @param array $definition
+     * @return PropertyDefinition
+     */
+    public function buildPropertyDefinition($name, array $definition)
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/PropertyDefinitionBuilder.php]: param `$name` could be strict, return type - also.

File: src/Paysera/Bundle/PhpGeneratorBundle/Twig/ApiMethodExtension.php [hunk 9]
```diff
@@ -7,6 +7,7 @@
 use Fig\Http\Message\StatusCodeInterface;
 use Paysera\Bundle\CodeGeneratorBundle\Entity\Definition\ApiDefinition;
 use Paysera\Bundle\CodeGeneratorBundle\Entity\Definition\ArgumentDefinition;
+use Paysera\Bundle\CodeGeneratorBundle\Entity\Definition\FilterTypeDefinition;
```
Comment (mSprunskas) [inline src/Paysera/Bundle/PhpGeneratorBundle/Twig/ApiMethodExtension.php]: not needed

**Comment by vbartusevicius** [comment]
We use annotation for `(datetime_timestamp)` - there is no other way to recognize timestamp without annotation

File: src/Paysera/Bundle/CodeGeneratorBundle/Service/PropertyDefinitionBuilder.php [hunk 10]
```diff
@@ -15,12 +15,7 @@ class PropertyDefinitionBuilder
         'time-only',
     ];
 
-    /**
-     * @param string $name
-     * @param array $definition
-     * @return PropertyDefinition
-     */
-    public function buildPropertyDefinition($name, array $definition)
+    public function buildPropertyDefinition(string $name, array $definition)
```
Comment (mSprunskas) [inline src/Paysera/Bundle/CodeGeneratorBundle/Service/PropertyDefinitionBuilder.php]: could be `public function buildPropertyDefinition(string $name, array $definition) : PropertyDefinition`

**Pull request merged**